### PR TITLE
fix dart environment sdk for flutter beta channel and some nullsafety issues

### PR DESCRIPTION
**Pull request by FluentChange**
simple improvement of dart version
and a null safety fix in example

### CODE REVIEW

File: example/pubspec.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ publish_to: none
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
```
Comment (xclud) [inline example/pubspec.yaml:18]: SDK version should be `2.12.0`. Please omit `-0`` part.

File: pubspec.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ homepage: https://pwa.ir
 repository: https://github.com/xclud/flutter_map
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
```
Comment (xclud) [inline pubspec.yaml:8]: The same here.

**Review by xclud — COMMENTED** [review]
@FluentChange Thank you for your contribution!

I see two small fixes are needed. Let's prepare for the next flutter version not just the preview. You may switch to flutter `master` channel if you cannot build with `2.12.0` (Comparing to 2.12.0-0).

BTW, for the example project, I hold the SDK version to 2.1.0 intentionally to show the developers they still use the package with earlier SDK versions. But anyway I welcome and appreciate your change. It's time to go 2.12.0. (Please let it be 2.12.0 as you suggested.)

File: pubspec.yaml (hunk 2 above)
Comment (FluentChange) [reply pubspec.yaml:8]: I think the the correct approach is with -0  
See migration and (un-)sound null safety here: https://dart.dev/null-safety/unsound-null-safety
But it's your project. And Flutter 2 is in stable today :-)

Comment (xclud) [reply pubspec.yaml:8]: Right. Flutter 2 and Dart 2.12 are out now officially. We can go for 2.12.0 now,

And no... it's not my project. It's open source and belongs to you as well :)

Comment (xclud) [reply pubspec.yaml:8]: The link you've posted is now updated and has removed the `-0` part.

**Pull request closed without merging**